actionview/test/template/form_with_helper_test.rb

```diff
@@ -0,0 +1,559 @@
+    expected = whole_form("an_url", method: "post") { '<label>Name</label>' }
+    assert_dom_equal expected, actual
+  end
+
+  def test_label_does_not_generate_for_attribute_when_given_nil
+    assert_tag_equal('<label>Title</label>') { |f| f.label(:title, for: nil) }
+    assert_tag_equal('<label>Title</label>') { |f| f.label(:title, class: nil) }
+  end
+
+  def test_label_with_attributes
+    assert_tag_equal('<label id="my_id">Title</label>') { |f| f.label(:title, id: "my_id") }
+    assert_tag_equal('<label id="my_id">Title</label>') { |f| f.label(:title, "id": "my_id") }
+    assert_tag_equal('<label for="my_for" id="my_id">Title</label>') { |f| f.label(:title, for: "my_for", id: "my_id") }
+    assert_tag_equal('<label for="my_for" id="my_id">Title</label>') { |f| f.label(:title, "for": "my_for", "id": "my_id") }
+  end
+
+  def test_label_with_block
+    assert_tag_equal('<label>The title, please:</label>') { |f| f.label(:title) { "The title, please:" } }
+  end
+
+  def test_label_with_block_and_html
+    assert_tag_equal('<label>Accept <a href="/terms">Terms</a>.</label>') { |f| f.label(:terms) { raw('Accept <a href="/terms">Terms</a>.') } }
+  end
+
+  def test_label_with_block_and_options
+    assert_tag_equal('<label for="my_for">The title, please:</label>') { |f| f.label(:title, for: "my_for") { "The title, please:" } }
+    assert_tag_equal('<label for="my_for">The title, please:</label>') { |f| f.label(:title, "for": "my_for") { "The title, please:" } }
+  end
+
+  def test_label_with_block_and_argument
+    I18n.with_locale :label do
+      assert_tag_equal('<label>Title</label>') { |f| f.label(:title) { |t| t } }
+    end
+  end
+
+  def test_label_with_block_in_erb
+    assert_dom_equal(
+      %{\n<label>\n<input name="post[title]" type="text" value="Babe went home">\n</label>},
+      view.render("test/field_with_label_with_block")
+    )
+  end
+
+  def test_file_field_has_no_size
+    assert_tag_equal('<input name="post[title]" type="file">') { |f| f.file_field(:title) }
+  end
+
+  def test_file_field_with_multiple_behavior
+    assert_tag_equal('<input multiple="multiple" name="post[attachment][]" type="file">') { |f| f.file_field(:attachment, "file", multiple: true) }
+  end
+
+  def test_file_field_with_multiple_behavior_and_explicit_name
+    assert_tag_equal('<input multiple="multiple" name="custom" type="file">') { |f| f.file_field(:blah, multiple: true, name: "custom") }
+    assert_tag_equal('<input multiple="multiple" name="custom[]" type="file">') { |f| f.file_field(:custom, multiple: true, scope: nil) }
+  end
+
+  def test_hidden_field
+    assert_tag_equal('<input name="post[title]" type="hidden" value="Catch 22">') { |f| f.hidden_field("title") }
+    assert_tag_equal('<input name="post[secret]" type="hidden" value="1">') { |f| f.hidden_field("secret?") }
+  end
+
+  def test_hidden_field_with_escapes
```

Comment:
Open all hidden files in the world on Oct 28, for 2 mins, estimated increase in the monthy divirce rate, 35-40%